driver-core/src/main/java/com/datastax/driver/core/SchemaParser.java

```diff
@@ -441,19 +441,29 @@ private void updateKeyspaces(
         metadata.triggerOnKeyspaceChanged(newKeyspace, oldKeyspace);
       }
       Map<String, TableMetadata> oldTables =
-          oldKeyspace == null ? new HashMap<String, TableMetadata>() : oldKeyspace.tables;
+          oldKeyspace == null
+              ? new HashMap<String, TableMetadata>()
+              : new HashMap<String, TableMetadata>(oldKeyspace.tables);
```

Comment:
Not the most elegant fix, but it has the merit of being straightforward.

I initially thought of replacing this with a simple `get` call:
```java
TableMetadata oldTable = oldTables.put(newTable.getName(), newTable);
```
But that doesn't work because during a partial refresh, `oldTables` points to the tables of the current keyspace.